packages/react/src/components/form/Search/SearchInput.tsx

```diff
@@ -0,0 +1,50 @@
+import { useMergeRefs } from '@floating-ui/react';
+import { type ChangeEvent, forwardRef, useRef, useState } from 'react';
+import { omit } from '../../../utilities';
+import { Input } from '../Input';
+
+export type SearchInputProps = Omit<
+  React.InputHTMLAttributes<HTMLInputElement>,
+  'readOnly' | 'type'
+> &
+  (
+    | { 'aria-label': string; 'aria-labelledby'?: never }
+    | { 'aria-label'?: never; 'aria-labelledby': string }
+  );
+
+export const SearchInput = forwardRef<HTMLInputElement, SearchInputProps>(
+  (props, ref) => {
+    const {
+      style,
+      defaultValue,
+      value,
+      onChange,
+      disabled,
+      className,
+      ...rest
+    } = props;
+    const inputRef = useRef<HTMLInputElement>();
+    const mergedRef = useMergeRefs([ref, inputRef]);
+
+    const [internalValue, setInternalValue] = useState(defaultValue ?? '');
+
+    const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
+      onChange?.(e);
+      value === undefined && setInternalValue(e.target.value);
+    };
+
+    return (
+      <Input
+        ref={mergedRef}
+        value={value ?? internalValue}
```

Comment:
De do not need internal value anymore? ☺️ 
can just pass `defaultValue` / `value` directly through `rest` <3
meaning we can also drop `ref` and `handleChange` 🫶

Reply:
dang, forgot to remove this before I set it to review